### PR TITLE
Use hardhat-exposed to reduce the need for mocks

### PR DESCRIPTION
**Pull request by Amxx**
#### PR Checklist

- [x] Tests

### CODE REVIEW

**Comment by frangio** [comment]
Inspected the above postinstall scripts and they're safe.

**Comment by frangio** [comment]
I would like us to review the "mock" name and switch to a new one. Maybe we can just rename everything to `Test` instead of `Mock`.

See https://github.com/OpenZeppelin/openzeppelin-contracts/issues/1320

**Comment by pcaversaccio** [comment]
FYI, just don't underestimate this breaking change - it will break many of my unit tests since I pull some of your mocks via submodules. So once you merge this PR to `master` and I update the submodules it will break (i.e. before you make a new release). Just wanted to quickly make you aware of this.

**Review by frangio — DISMISSED** [review]
Ready to merge this.

File: contracts/utils/Base64.sol [hunk 1]
```diff
@@ -12,7 +12,7 @@ library Base64 {
     /**
      * @dev Base64 Encoding/Decoding Table
      */
-    string internal constant _TABLE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+    string private constant _TABLE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
```
Comment (frangio) [inline contracts/utils/Base64.sol:15]: Actually, I realized I had this pending comment. We should do this in a separate PR if we want to do it.

```suggestion
    string internal constant _TABLE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
```

File: contracts/mocks/ERC20Mock.sol [hunk 2]
```diff
@@ -1,33 +1,16 @@
 // SPDX-License-Identifier: MIT
-
 pragma solidity ^0.8.0;
 
-import "../token/ERC20/ERC20.sol";
+import {ERC20} from "../token/ERC20/ERC20.sol";
```
Comment (Amxx) [inline contracts/mocks/ERC20Mock.sol:4]: I believe that is the first time we use this syntax. Is that something we want to be doing more ?

File: contracts/utils/Base64.sol (hunk 1 above)
Comment (Amxx) [reply contracts/utils/Base64.sol:15]: ok, lets make another PR

**Comment by Amxx** [comment]
LGTM

File: contracts/mocks/ERC20Mock.sol (hunk 2 above)
Comment (frangio) [reply contracts/mocks/ERC20Mock.sol:4]: Not sure. It wasn't a conscious decision, I think I went along with the syntax from `a16z/erc4626-tests`.

**Review by frangio — APPROVED** [review]
